cli/utils.go

```diff
@@ -92,3 +94,42 @@ func parseBillingAddress(address string) (*apppb.BillingAddress, error) {
 		Zipcode:       strings.Trim(splitAddress[4], " "),
 	}, nil
 }
+
+func saveLogsToFile(filePath, format string, logs []string) error {
+	// Ensure the directory exists
+	dir := filepath.Dir(filePath)
+	if err := os.MkdirAll(dir, 0o750); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("could not create directory: %s", dir))
+	}
+
+	// Validate directory permissions
+	info, statErr := os.Stat(dir)
+	if statErr != nil {
+		return errors.Wrap(statErr, fmt.Sprintf("could not stat directory: %s", dir))
+	}
+
+	if !info.IsDir() {
+		return fmt.Errorf("resolved path is not a directory: %s", dir)
+	}
+
+	var output []byte
+	switch format {
+	case "json":
+		var err error
+		output, err = json.MarshalIndent(logs, "", "  ")
```

Comment:
When outputting data as json, I think it's customary to output:
```
{"ts": 123, "message": "log line 1"}
{"ts": 124, "message": "log line 2"}
```

IIUC, this will instead output an array with 2 items:
```
[
  {"ts": 123, "message": "log line 1"},
  {"ts": 124, "message": "log line 2"}
]
```

Related to the above, the reason is because it is often difficult for json parsers/utilities to work with large lists efficiently/ergonomically.

This feels like a recipe to OOM. I've downloaded multi-GB log files